accord-core/src/main/java/accord/local/KeyHistory.java

```diff
@@ -48,7 +49,33 @@ public enum KeyHistory
     /**
      * Load recovery information for all keys into memory before processing the command.
      */
-    RECOVER,
+    RECOVER;
+    
+    public boolean satisfiesIfPresent(KeyHistory that)
+    {
+        return satisfies(that, ASYNC);
+    }
+
+    public boolean satisfies(KeyHistory that)
+    {
+        return satisfies(that, SYNC);
+    }
+
+    private boolean satisfies(KeyHistory that, KeyHistory ifSyncRequireAtLeast)
+    {
+        switch (that)
+        {
+            default: throw new AssertionError("Unhandled KeyHistory: " + that);
+            case NONE:
+                return true;
+            case RECOVER:
+            case TIMESTAMPS:
+                return this == that;
+            case ASYNC:
+            case INCR:
+            case SYNC:
+                return this.compareTo(ifSyncRequireAtLeast) >= 0;
```

Comment:
so for `satisfies` (only method used in this PR) this is only true for `SYNC` and `RECOVER`?  `ASYNC`, `INCR`, and `SYNC` are all new to me, so need to catch up on the recent changes to better understand this change.

Reply:
yes, ASYNC and INCR basically don't guarantee they have fetched what was requested. They'll accept partial information. That said, this isn't implemented on the load side yet so today it's only declarative of intent/requirements on the task side.